docs/source/getting_started/writing_code.ipynb

```diff
@@ -0,0 +1,308 @@
+    "\n",
+    "First, we recommend installing `mamba` into the base environment. This will take a few minutes (especially solving the environment), but only has to be done once (per running workspace session) and will make future environment creation and changes much faster!\n",
+    "\t\n",
+    "Go to File -> New -> Terminal to open up a terminal.\n",
+    "\n",
+    "```\n",
+    "conda install -c conda-forge mamba\n",
+    "```\n",
+    "\n",
+    "If you have installed `mamba`, replace all `conda` calls below with `mamba` calls for much faster execution!\n",
+    "\n",
+    "NOTE: Because modifications to the base conda environment are reset after every workspace restart, you will have to re-install mamba into the base environment after every workspace restart. Future iterations of MAAP will have mamba installed in the base environment, but we don’t have access to those updates yet. However, you do not need to have mamba installed to activate conda environments you have already created (even if they were created using mamba), so you only need to (re-)install mamba when you need to create or modify conda environments.\n",
+    "\n",
+    "The recommended approach is to use environment YML files. For example, create a file like the following called `newenv2.yml`:\n",
+    "\n",
+    "newenv2.yml\n",
+    "```\n",
+    "name: newenv2\n",
+    "channels:\n",
+    "\t- conda-forge\n",
+    "\t- defaults\n",
+    "dependencies:\n",
+    "\t- mamba\n",
+    "\t- python=3.8\n",
+    "\t- ipykernel\n",
+    "\t- pandas\n",
+    "\t- geopandas\n",
+    "```\n",
+    "\n",
+    "Then, from a terminal, you can create this environment with a command like: \n",
+    "```\n",
+    "conda env create -f newenv2.yml\n",
+    "``` \n",
+    "This command may take a few minutes to run and may appear to hang on “Solving Environment” (depending on the complexity of your environment). However, in most cases, it should eventually solve the environment and complete the installation process.\n",
+    "\n",
+    "NOTE: Custom environments are deleted, and modifications to the base conda environment (e.g., installing mamba) are reset, when the workspace is Stopped. To preserve a custom environment after the workspace shutdowns, create the environment that is saved to the home directory instead of the default system directory for environments. Make sure to not make a conda env inside a git-tracked folder, or if you do add it to the `.gitignore`. If git is tracking an env, it could cause your workspace to crash. This can be done via the following commands, to create the environment at the path `./myenvs/testenv` with the flag `-p` and to activate it using the full path:\n",
+    "```\n",
+    "mamba env create -f env.yml -p ./myenvs/testenv\n",
+    "conda  activate ./myenvs/testenv\n",
```

Comment:
Should we change this line to `conda env create -f newenv2.yml -p ./myenvs/testenv` to show the user it can also be run with conda (we told them above they can just replace conda with mamba). Also, we should continue calling the file newenv2.yml to make it easier to copy and paste

Overall, I am a little confused about the NOTE:. I think we should split this up into 2 different paragraphs. One related to custom environment persistence, and another related to adding it to the .gitignore because this current way makes it seem like those 2 commands will add the conda env to .gitignore
I suggest:
"NOTE: Custom environments are deleted, and modifications to the base conda environment (e.g., installing mamba) are reset, when the workspace is stopped. To preserve a custom environment after the workspace shutdowns, create the environment that is saved to the home directory instead of the default system directory for environments.
```
conda env create -f newenv2.yml -p ./myenvs/testenv
conda  activate ./myenvs/testenv
```
Make sure that the file path in this command for the conda env is not inside a git-tracked folder, or if you do add it to the `.gitignore`. If git is tracking an env, it could cause your workspace to crash. `.gitignore` goes in the root directory of your git repository, and you can just add the file name to this list"

Note that I added information about how to add to a .gitignore because the average user might not be familiar with .gitignore. Even I am not sure what they are adding to the .gitignore. When I ran those commands, no file was created in my myenvs folder. What should they be adding to the .gitignore?

UPDATE: I talked to Rob about this and he thinks the .yml is what we are suggesting not to add to to git, and we should find a best practice for this

Reply:
From Anil regarding this: If you are referring to our workspaces? Yes. We have mamba installed in all of them to speed up package installation.